src/stage1/forward.jl

```diff
@@ -118,6 +118,13 @@ function (::∂☆internal{1})(args::AbstractTangentBundle{1}...)
     end
 end
 
+function ChainRulesCore.frule_via_ad(::DiffractorRuleConfig, partials, args...)
+    tangents = map(partials, args) do p, a
+        TangentBundle{1}(a, (p,))
+    end
+    ∂☆internal{1}()(tangents...)
+end
```

Comment:
I don't see what's going wrong here. The error is this:
```
julia> z45, delta45 = frule_via_ad(DiffractorRuleConfig(), (ZeroTangent(),1), sqrt, 2.0)
ERROR: MethodError: no method matching iterate(::Diffractor.TangentBundle{1, Float64, Tuple{Float64}})
Closest candidates are:
  iterate(::Union{LinRange, StepRangeLen}) at range.jl:737
  iterate(::Union{LinRange, StepRangeLen}, ::Int64) at range.jl:737
  iterate(::T) where T<:Union{Base.KeySet{<:Any, <:Dict}, Base.ValueIterator{<:Dict}} at dict.jl:695
  ...
Stacktrace:
 [1] indexed_iterate(I::Diffractor.TangentBundle{1, Float64, Tuple{Float64}}, i::Int64)
   @ Base ./tuple.jl:92
 [2] top-level scope
   @ REPL[54]:1
```
It certainly calls `shuffle_base(r) = TangentBundle{1}(r[1], (r[2],))` (10 lines up) but after that I don't know.

Reply:
I don't think you can destructure a Tangent Bundle like this.

Reply:
But what is destructuring it? The thing I construct called `tangents` is just a Tuple, and looks acceptable to `∂☆internal{1}()`. From the type `tangents[2]` is being iterated but I don't see where.

Reply:
My suspicion is that it's you doing `z45, delta45 = frule_via_ad(...)` inside the test, but I could be wrong.

Reply:
Oh right, ok. That would explain the short stack trace! This must return a tuple but surely I can figure out how to convert it.